examples/erc1155/lib.rs

```diff
@@ -0,0 +1,687 @@
+        /// be assigned to the caller (a.k.a the minter).
+        ///
+        /// Note that as implemented anyone can mint tokens. If you were to deploy this to a
+        /// production environment you'd probably want to lock down the addresses that are allowed
+        /// to mint tokens.
+        #[ink(message)]
+        pub fn mint(&mut self, token_id: TokenId, value: Balance, _data: Vec<u8>) {
+            assert!(
+                token_id <= self.token_id_nonce,
+                "The `token_id` {:?} has not yet been created in this contract.",
+                token_id
+            );
+
+            self.balances.insert((self.env().caller(), token_id), value);
+
+            // Emit transfer event but with mint semantics
+            self.env().emit_event(TransferSingle {
+                operator: self.env().caller(),
+                from: AccountId::from([0; 32]),
+                to: self.env().caller(),
+                token_id,
+                value,
+            });
+        }
+
+        // Helper function for performing single token transfers.
+        //
+        // Should not be used directly since it's missing certain checks which are important to the
+        // ERC-1155 standard (it is expected that the caller has already perfomred these).
+        fn perform_transfer(
+            &mut self,
+            from: AccountId,
+            to: AccountId,
+            token_id: TokenId,
+            value: Balance,
+            data: Vec<u8>,
+        ) {
+            assert!(
+                self.balance_of(from, token_id) >= value,
+                "Insufficent token balance for transfer."
+            );
+
+            if let Some(b) = self.balances.get_mut(&(from, token_id)) {
+                *b -= value
+            }
+
+            self.balances
+                .entry((to, token_id))
+                .and_modify(|b| *b += value)
+                .or_insert(value);
+
+            self.env().emit_event(TransferSingle {
+                operator: self.env().caller(),
+                from,
+                to,
+                token_id,
+                value,
+            });
+
+            // Quick Haxx, otherwise my tests just panic due to the use of eval_contract()
+            #[cfg(not(test))]
+            {
+                // If our recipient is a smart contract we need to see if they accept or
+                // reject this transfer. If they reject it we need to revert the call.
+                let params = build_call::<ink_env::DefaultEnvironment>()
+                    .callee(to)
+                    .gas_limit(5000) // Q: What's the correct amount to use here?
+                    .exec_input(
+                        ExecutionInput::new(Selector::new(MAGIC_VALUE))
+                        .push_arg(self.env().caller())
+                        .push_arg(from)
+                        .push_arg(token_id)
+                        .push_arg(value)
+                        .push_arg(data)
```

Comment:
I really don't want to be that guy but I wonder why `rustfmt` CI does not want another indentation here.

Reply:
The word of `rustfmt` is law 🤷

Reply:
Still better than not using `rustfmt` to start with ...